packages/core/src/App/Containers/AccountSwitcher/account-switcher.jsx

```diff
@@ -1208,6 +1212,43 @@ const AccountSwitcher = props => {
     const real_account = props.is_pre_appstore ? traders_hub_real_accounts : default_real_accounts;
 
     const demo_accounts = props.is_pre_appstore ? traders_hub_demo_account : default_demo_accounts;
+    const first_real_login_id = props.account_list?.find(account => account.loginid?.[0]).loginid;
+
+    const TradersHubRedirect = () => {
+        const TradersHubLink = () => {
+            const handleRedirect = () => {
+                if (!props.is_virtual && isDemoAccountTab) {
+                    props.switchAccount(props.virtual_account_loginid);
+                } else if (props.is_virtual && isRealAccountTab) {
+                    props.switchAccount(first_real_login_id);
+                }
+                history.push(routes.traders_hub);
+                props.setTogglePlatformType('cfd');
+            };
+
+            return (
+                <>
+                    <div className='acc-switcher__traders-hub'>
+                        <BinaryLink onClick={handleRedirect} className='acc-switcher__traders-hub--link'>
+                            <Text size='xs' align='center' className='acc-switcher__traders-hub--text'>
+                                <Localize i18n_default_text="Looking for CFD accounts? Go to Trader's hub" />
+                            </Text>
+                        </BinaryLink>
+                    </div>
+                    <div className='acc-switcher__separator' />
+                </>
+            );
+        };
+
+        if (props.is_pre_appstore) {
+            if (isRealAccountTab && props.has_any_real_account) {
+                return <TradersHubLink />;
+            } else if (isDemoAccountTab) {
+                return <TradersHubLink />;
+            }
+        }
+        return null;
```

Comment:
```suggestion
        if (props.is_pre_appstore) {
            if (isRealAccountTab && props.has_any_real_account) {
                return <TradersHubLink />;
            }
            return <TradersHubLink />;
        }
        return null;
```

Reply:
I would have to check the demo as well, if not then it will still show for real tab if the user doesnt have an account